test/test_optim.py

```diff
@@ -343,6 +349,8 @@ def test_adagrad(self):
                 self._build_params_dict(weight, bias, lr=1e-2),
                 lr=1e-1)
         )
+        with self.assertRaisesRegex(ValueError, "Invalid lr_decay value: -0.5"):
+            optim.Adagrad(None, lr=1e-2, lr_decay=-0.5)
 
     def test_adagrad_sparse(self):
         self._test_rosenbrock_sparse(
```

Comment:
for `test_adamax`below, some test cases use `optim.Adagrad` instead of `optim.Adamax`. Is this intentional? Or a bug?

Reply:
Sounds like a bug to me.

Reply:
I will address this in a separate PR